scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/InProcessTaskExecutor.java

```diff
@@ -208,10 +209,16 @@ private void saveScriptAsFile(String path, Script<?> script, TaskContext taskCon
 
         String code = script.fetchScript().trim();
 
-        //Check if the path is an absolute path or a relative path, if it is a relative path store the file in the local space
         Path p = Paths.get(path);
+
+        //Check if the path is an absolute path or a relative path, if it is a relative path store the file in the local space
         boolean isAbsolute = p.isAbsolute();
         if (isAbsolute) {
+            //Check if the parent path exists, if not create it
+            Path hasParent = p.getParent();
+            if(!Files.exists(hasParent)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this with a call to the "toFile().exists()" method [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3725)